beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/multipeer/chains/SyncSourceFactory.java

```diff
@@ -47,7 +48,7 @@ public SyncSource getOrCreateSyncSource(final Eth2Peer peer, final Spec spec) {
             .getConfig()
             .toVersionDeneb()
             .map(SpecConfigDeneb::getMaxBlobsPerBlock)
-            .orElse(0);
+            .orElse(MAX_BLOBS_PER_BLOCK);
```

Comment:
I think one of the things we target now is to reduce number of constants, especially while this one should come from config. I think something like this could be better `final int initMaxBlobs = Optional.ofNullable(spec.forMilestone(SpecMilestone.DENEB)).map(specDeneb -> specDeneb.getConfig().toVersionDeneb().orElseThrow().getMaxBlobsPerBlock()).orElse(0);
` (not tested)

Reply:
A slight simplification:

Reply:
@mehdi-aouadi this way we will not change this number in the next fork if it's changed. The idea was to leave `spec.atEpoch()` thing but do a fallback not on 0, on Deneb number instead.

Reply:
Now I understand what you meant. Added the sspec.atEpoch(peer.finalizedEpoch())` and used your suggestion to calculated the fallback value